pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go

```diff
@@ -32,6 +39,7 @@ type NMState struct{}
 
 // +k8s:openapi-gen=true
 type KubeMacPool struct {
+	CertificateKnobs
```

Comment:
I think we should unify all cert settings under a single root-level structure (like imagePullPolicy) instead of making it specific for each component. Any reason not-to?

Reply:
Are we going to need to set different values per component ?

Reply:
Not under HCO. I'd start with single top-level settings. If somebody comes up with a reason to have those values different per each component, we can make it more granular.

Reply:
One issue I see is for example kmp webhooks are critical but not the case for knmstate, in case we find that there are some downtime during rotation we can increase rotation at kmp and not at knmstate, what do you think ?

Reply:
Good point. Let's think how it will be used.

The user will set some HCO CR. That will be propagated to CNAO CR, to one ore multiple knobs and then down to components.

If the scenario you describe will happen, there will be still only a single set of knobs on HCO, so if we want to change the config to prefent some bug from happening, we would need to hardcode the special value ourselves in HCO or CNAO.

At that point, granular control would make it slightly easier, but is not a must.

Reply:
Ack I will make it common.